polaris/hub/client.py

```diff
@@ -174,14 +174,16 @@ def _base_request_to_hub(self, url: str, method: str, **kwargs):
             if response_status_code == 500:
                 raise
 
-            # If not an internal server error, the hub should always return a JSON response
-            # with additional information about the error.
-            response = response.json()
-            response = json.dumps(response, indent=2, sort_keys=True)
+            # If JSON is included in the response body, we retrieve it and format it for output. If not, we retrieve
+            # the string representation of the response body and move forward
+            try:
+                response = response.json()
+                response = json.dumps(response, indent=2, sort_keys=True)
+            except (json.JSONDecodeError, TypeError):
```

Comment:
Given that the Hub should always return valid JSON, we shouldn't need to worry about a `TypeError` in the case `json.dumps(...)` fails. However, I added it here to be sure. I also like the readability aspect of directly specifying which errors _could_ happen.

Minor, but let me know if anyone has thoughts.

Reply:
Out of curiosity – What prompted this change?

Reply:
If a user defines a benchmark split with >150,000 rows and tries to upload it to the hub, our infra provider will deny the request because the payload is too large. 

In their response, there will be no JSON and so the user will get a very confusing error message related to a JSONDecodeError. This change prevents that.

Reply:
I see. How about we add a comment for future reference to make it clear why we're handling these exceptions?

```python
try:
    response_json = response.json()
    response = json.dumps(response_json, indent=2, sort_keys=True)
except (json.JSONDecodeError, TypeError):
    # If the response is not valid JSON or if json.dumps fails, fallback to plain text.
    # This is important for handling large payloads that our infra provider may reject,
    # resulting in a non-JSON error response.
    response = response.text
```

Reply:
You don't think the comment on lines 177-178 is descriptive enough?

I'm also not sure if mentioning the limitations of our infra provider in the codebase is advisable. What do you think? 

If we feel strongly, I'll add something a bit more descriptive.

Reply:
> You don't think the comment on lines 177-178 is descriptive enough?
The comment on lines 177-178 is helpful, but I felt it lacked some context, which is why I asked about the reason behind the changes.

> I'm also not sure if mentioning the limitations of our infra provider in the codebase is advisable. What do you think?
I agree.

As long as we remember why we added this, that's what matters most.